### PR TITLE
remove backwards compatibility

### PR DESCRIPTION
**Pull request by FoamyGuy**
This change would go along with: https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/pull/167. They would need to be merged at the same time to keep things working I think.

If that ESP32SPI PR is merged it will make the sockets share the same API so I think this backwards compatibility behavior is no longer needed. 

The FakeSSLSocket does need to have a recv_into binding in order for it to actually work with this removed. 

Using the changes from this PR and the changes from ESP32SPI PR together on a PyPortal I am able to successfully `wget()` the image file without any stair stepping chunked issues.

### CODE REVIEW

**Comment by FoamyGuy** [comment]
I'm not entirely sure how to re-create the failed test that actions is failing for locally. I tried running the test locally but it seems to pass:
image

It seems like the tests failing are for legacy compatible behavior. If the behavior that is referring to is the stuff being removed by https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/pull/167 then perhaps we don't need to keep the tests for it since the socket APIs will match after that if I understand correctly.

If we do need to keep the legacy tests, it looks like it will need to have recv_into() added to the mocket in legacy_modcket.py. Similar to how it was added to FakeSSLContext I guess. But I'm not really sure how to implement it in the mocket, I could create a function for it, but I don't know what would need to go inside of it.

**Comment by FoamyGuy** [comment]
The latest commit removes the legacy tests which were validating behavior related to the non-standard socket behavior from ESP32SPI. https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/pull/167 will make the socket API more standardized and matching with CPython, so these tests are no longer needed.

I think this is ready for review.


I tested these changes successfully in the following contexts:
- On a PyPortal 8.2.0-rc.0 with the simpletest example from this repo
- On a Feather ESP32-S3 + Ethernet Feathering with the ethernet_simpletest example from this repo
- These learn guide projects:
    - https://learn.adafruit.com/pyportal-github-stars-trophy (uses pyportal library)
    - https://learn.adafruit.com/pyportal-weather-station/
    - https://learn.adafruit.com/pyportal-smart-thermometer-with-analog-devices-adt7410-adafruit-io-and-circuitpython
    - https://learn.adafruit.com/cleveland-museum-of-art-pyportal-frame (this is the one that had problems with the "chunky" downloads. It now works correctly)

**Review by tannewt — APPROVED** [review]
Thank you! Not merging so you can time it right.

File: adafruit_requests.py [hunk 1]
```diff
@@ -217,12 +215,6 @@ def __exit__(
         self.close()
 
     def _recv_into(self, buf: bytearray, size: int = 0) -> int:
-        if self._backwards_compatible:
-            size = len(buf) if size == 0 else size
-            b = self.socket.recv(size)
-            read_size = len(b)
-            buf[:read_size] = b
-            return read_size
         return cast("SupportsRecvInto", self.socket).recv_into(buf, size)
```
[the remark this answers is not in the document]
Comment (FoamyGuy) [reply adafruit_requests.py]: Thank you. The latest commits remove `LegacyCircuitPythonSocketType` and the cast here, which was the only usage of `cast()` so I was able to remove that declaration / import as well.

I successfully re-tested with these changes using PyPortal 8.2.0-rc.1, a Feather ESP32S3 TFT 8.2.0-rc.0, and a Metro M7 Airlift 8.2.0-rc.1 and the respective simpletest scripts that run on them.

[the remark this answers is not in the document]
Comment (jepler) [reply adafruit_requests.py]: thank you! it is nice to be getting rid of the cruft.

**Pull request merged**